src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -82,6 +84,11 @@ private static bool InUnsafeMethod(Symbol symbol)
         private Dictionary<LocalSymbol, (uint RefEscapeScope, uint ValEscapeScope)>? _localEscapeScopes;
         private Dictionary<BoundValuePlaceholderBase, uint>? _placeholderScopes;
         private uint _patternInputValEscape;
+#if DEBUG
+        private const int MaxTrackVisited = 100; // Avoid tracking if too many expressions.
```

Comment:
Just curious, why put this limit?

Reply:
I was concerned about the overhead of creating large dictionaries for verification, for instance in unit tests with long array initializers.